### PR TITLE
HW 04.03

### PR DESCRIPTION
**Pull request by kirylshreyter**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,13 +9,20 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".activities.MainActivity">
+
+        <!--Application activities-->
```
Comment (ilya-shknaj) [inline app/src/main/AndroidManifest.xml]: I think this comment can be removed

File: app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+public class AboutActivity extends BaseActivity {
+    private IntentFilter filter = new IntentFilter(Actions.CUSTOM_ACTION);
+    private ServiceConnection connection = new AboutServiceConnection(this);
+    private Intent aboutServiceIntent;
+    private BroadcastReceiver receiver;
+    private Fragment aboutFragment = new AboutFragment();
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```
Comment (ilya-shknaj) [inline app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java:34]: on create method is too big and contain a lot of logic. Please split it to few separated methods

File: app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_about);
+
+        if (savedInstanceState != null) {
+            aboutFragment = getSupportFragmentManager().getFragment(savedInstanceState, Constants.ABOUT_FRAGMENT_KEY);
```
Comment (ilya-shknaj) [inline app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java]: please describe what this line of code doing?

File: app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                    .beginTransaction()
+                    .add(R.id.daily_digest, aboutFragment)
+                    .commit();
+        }
+
+        aboutServiceIntent = new Intent(this, AboutService.class);
+        bindService(aboutServiceIntent, connection, BIND_AUTO_CREATE);
+        registerReceiver();
+
+        if (savedInstanceState == null) {
```
Comment (ilya-shknaj) [inline app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java]: why you need so strange logic if savedInstanceState==null run service?

File: app/src/main/java/com/kirylshreyter/diamondapp/components/fragments/AboutFragment.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    private void loadFragmentState(Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            TextView text;
+            ProgressBar progressBar;
+            try {
+                text = getActivity().findViewById(R.id.about_activity_text_view);
+                progressBar = getActivity().findViewById(R.id.about_activity_progress_bar);
+            } catch (NullPointerException exception) {
```
Comment (ilya-shknaj) [inline app/src/main/java/com/kirylshreyter/diamondapp/components/fragments/AboutFragment.java]: it's wrong catch nullPointerException. You need to remove it and fix root of issue if nullPointerException happening

File: app/src/main/java/com/kirylshreyter/diamondapp/components/services/AboutService.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.kirylshreyter.diamondapp.common.Actions;
+import com.kirylshreyter.diamondapp.common.Constants;
+import com.kirylshreyter.diamondapp.common.Tags;
+import com.kirylshreyter.diamondapp.components.binders.AboutBinder;
+import com.kirylshreyter.diamondapp.services.IBaseService;
+import com.kirylshreyter.diamondapp.services.IServiceResult;
+import com.kirylshreyter.diamondapp.services.impl.FetchService;
+
+@WorkerThread
```
Comment (ilya-shknaj) [inline app/src/main/java/com/kirylshreyter/diamondapp/components/services/AboutService.java]: service start in UI thread. It's wrong mark it by WorkerThread annotation

File: app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java (hunk 4 above)
Comment (kirylshreyter) [reply app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java]: just wanted to start service once on first run of activity, but the service starts every time when, for example, phone orientation has changed.

File: app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java (hunk 3 above)
Comment (kirylshreyter) [reply app/src/main/java/com/kirylshreyter/diamondapp/activities/AboutActivity.java]: this line of code assign fragment to variable for further processing

**Pull request merged**